src/theory/bv/proof_checker.cpp

```diff
@@ -0,0 +1,54 @@
+/*********************                                                        */
+/*! \file proof_checker.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Mathias Preiner
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Implementation of bit-vectors proof checker
+ **/
+
+#include "theory/bv/proof_checker.h"
+
+namespace CVC4 {
+namespace theory {
+namespace bv {
+
+BVProofRuleChecker::BVProofRuleChecker() : d_bb(new BBSimple(nullptr)) {}
```

Comment:
I'm wondering if it would be better to make d_bb a local variable wheren BV_BITBLAST is.  This would make the proof checker stateless (and hence more trustworthy conceptually).
Is there a reason for having d_bb be a member? E.g. do you need its cache?

Reply:
The cache is not mandatory and it also works if we make the bitblaster local in the BV_BITBLAST rule. If we ever run into performance issues because of the cache, we can always fix it.